src/structures/guild_preference.ts

```diff
@@ -76,49 +76,57 @@ export default class GuildPreference {
     /** The GuildPreference's respective GameOptions */
     private gameOptions: GameOptions;
 
-    constructor(guildID: string, json?: GuildPreference) {
+    constructor(guildID: string, options?: GameOptions) {
         this.guildID = guildID;
+        this.gameOptions = options || { ...DEFAULT_OPTIONS };
+    }
+
+    static fromGuild(guildID: string, json?: GuildPreference): GuildPreference {
+        let gameOptions: GameOptions;
         if (!json) {
-            this.gameOptions = { ...DEFAULT_OPTIONS };
-            return;
+            gameOptions = { ...DEFAULT_OPTIONS };
+            return new GuildPreference(guildID, gameOptions);
         }
-        this.gameOptions = json.gameOptions;
+        // eslint-disable-next-line prefer-destructuring
+        gameOptions = json.gameOptions;
         // apply default game option for empty
         let gameOptionModified = false;
         for (const defaultOption in DEFAULT_OPTIONS) {
-            if (!(defaultOption in this.gameOptions)) {
-                this.gameOptions[defaultOption] = DEFAULT_OPTIONS[defaultOption];
+            if (!(defaultOption in gameOptions)) {
+                gameOptions[defaultOption] = DEFAULT_OPTIONS[defaultOption];
                 gameOptionModified = true;
             }
         }
 
         // extraneous keys
-        for (const option in this.gameOptions) {
+        for (const option in gameOptions) {
             if (!(option in DEFAULT_OPTIONS)) {
-                delete this.gameOptions[option];
+                delete gameOptions[option];
                 gameOptionModified = true;
             }
         }
+        const guildPrefrence = new GuildPreference(guildID, gameOptions);
         if (gameOptionModified) {
-            this.updateGuildPreferences();
+            guildPrefrence.updateGuildPreferences();
         }
+        return guildPrefrence;
     }
 
     /**
      * Sets the limit option value
      * @param limit - The limit range value
      */
-    setLimit(limitStart: number, limitEnd: number) {
+    async setLimit(limitStart: number, limitEnd: number) {
```

Comment:
This usually isn't a problem in normal gameplay because the time between a user setting a game option and the time a new game or round starts is in the magnitude of seconds. In a test where the following line uses the `guildPreference`, it matters